components/autocomplete/index.js

```diff
@@ -176,40 +351,76 @@ class Autocomplete extends Component {
 		// Any handled keycode should prevent original behavior. This relies on
 		// the early return in the default case.
 		event.preventDefault();
-		event.stopImmediatePropagation();
+		event.stopPropagation();
+	}
+
+	toggleKeyEvents( isListening ) {
+		// This exists because we must capture ENTER key presses before Editable.
+		// It seems that react fires the simulated capturing events after the
+		// native browser event has already bubbled so we can't stopPropagation
+		//  and avoid Editable getting the event from TinyMCE, hence we must
+		// register a native event handler.
+		if ( ! this.node ) {
+			return;
+		}
+		const realNode = findDOMNode( this.node );
+		const handler = isListening ? 'addEventListener' : 'removeEventListener';
+		realNode[ handler ]( 'keydown', this.setSelectedIndex, true );
+	}
+
+	componentWillMount() {
+		const { completers } = this.props;
+		// Some completers must do asynchronous requests to get their options so
+		// we calculate that here.
+		completers.forEach( ( { getOptions }, idx ) => {
+			getOptions().then( ( options ) => {
+				this.allOptions[ idx ] = options.map( ( option, key ) => ( { ...option, key } ) );
+				this.forceUpdate();
```

Comment:
Rather than forcing the update, should we be setting all options into state via `setState`?

Reply:
Yeah I probably should. I did it this way because I was too lazy to rewrite the reset function (which would have deleted the cache the way it is currently written).

Reply:
I have moved the options into state.